Sprint-3/3-mandatory-practice/implement/repeat.js

```diff
@@ -1,5 +1,10 @@
-function repeat() {
-    return "hellohellohello";
+function repeat(str, rep) {
+  myStr = "";
+  for (let i = 0; i < rep; i++) {
+    myStr = myStr + str;
+  }
+  if (rep < 0) return "negative counts are not valid";
+  return myStr;
```

Comment:
```suggestion
function repeat(str, rep) {
  if (rep < 0) return "negative counts are not valid";
  myStr = "";
  for (let i = 0; i < rep; i++) {
    myStr = myStr + str;
  }
  return myStr;
```

You should always be validating your inputs first, that way there are no dubious values being passed down your program which can create a problem. In this case you should check first that the count is a positive integer, before running any other part of your function.

Reply:
thanks, I moved the filter if(rep<0) to the first line of the function to skip out fast when it doesn't match.